godot-macros/src/class/data_models/func.rs

```diff
@@ -167,6 +169,19 @@ fn make_forwarding_closure(class_name: &Ident, signature_info: &SignatureInfo) -
                 }
             }
         }
+        ReceiverType::GdSelf => {
+            quote! {
+                |instance_ptr, params| {
+                    let ( #(#params,)* ) = params;
+
+                    let storage =
+                        unsafe { ::godot::private::as_storage::<#class_name>(instance_ptr) };
+                    let gd_self: ::godot::obj::Gd<#class_name> = storage.get().base.share().cast();
```

Comment:
This relies on `base` to create the `Gd<T>`. If there's a way to go from the `InternalStorage` to the `Gd` that doesn't require there to be a `base` that would be more generic.
If there isn't, then at least getting the field tagged `base` rather than assuming it's named `base` would be important.

Reply:
i tried implementing this and i just added a new field to `InstanceStorage` that contained a `Base`, and i think that's a better approach to take at least.